runtime/common/metering.go

```diff
@@ -425,11 +425,18 @@ func BigIntByteLength(v *big.Int) int {
 //
 
 func NewPlusBigIntMemoryUsage(a, b *big.Int) MemoryUsage {
-	// max(|a|, |b|) + 5
+	// if |a|==|b|==0, then 0
+	// else max(|a|, |b|) + 5
+	aWordLength := len(a.Bits())
+	bWordLength := len(b.Bits())
+
+	if aWordLength == 0 && bWordLength == 0 {
+		return NewBigIntMemoryUsage(0)
+	}
```

Comment:
I ignored the 0 cases, because I understood you as saying that the library always pre-allocates some words, but this does not hurt 👍

Reply:
You're right, the lib over allocates but only in the general cases. 
We can add these edge cases anytime later if you think we should "tighten" the metering.